### PR TITLE
Fix 4303 output layer metadata

### PR DESCRIPTION
**Pull request by ismailsunni**
### What does it fix?
* Ticket: #4303 
* Funded by: DFAT
* Description: 
   - Adding output layer URI in the output layer metadata (e.g. aggregate_hazard_impacted in the image below)
screen shot 2017-07-26 at 22 35 12

### CODE REVIEW

File: safe/gui/widgets/dock.py [hunk 1]
```diff
@@ -915,7 +914,7 @@ def layer_changed(self, layer):
             if show_keywords:
                 if inasafe_keyword_version_key not in keywords.keys():
                     show_keyword_version_message(
-                        self, 'No Version', self.inasafe_version)
```
Comment (Gustry) [inline safe/gui/widgets/dock.py]: Is this block still valid? I think this was legacy and I was wondering if this block can be reached again today

**Comment by ismailsunni** [comment]
Oh no, the travis is failed, and the value to compare is depending on datetime in the output layer path.

File: safe/gui/widgets/dock.py (hunk 1 above)
Comment (ismailsunni) [inline safe/gui/widgets/dock.py]: It's still possible if you have corrupted metadata xml file that doesn't have `keyword version` in it. I prefer to keep this althought the probability is very small.

Comment (Gustry) [inline safe/gui/widgets/dock.py]: ok

**Comment by Gustry** [comment]
Can't you check that the expected provenance is a subset of the real provenance? IE not checking that dict are equals, but it's a subset.

**Comment by ismailsunni** [comment]
> Can't you check that the expected provenance is a subset of the real provenance? IE not checking that dict are equals, but it's a subset.

Yes, I did it. I also check if the key for output layers exist

**Pull request merged**